targets/arduino_101/Makefile.arduino_101

```diff
@@ -0,0 +1,183 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+.DEFAULT_GOAL := all
+
+ifeq ($(.DEFAULT_GOAL),)
+  $(warning no default goal is set)
+endif
+
+ifeq ($(MAKECMDGOALS),qemu)
+BOARD ?= qemu_x86
+else
+BOARD ?= arduino_101
+BOARD_NAME ?= arduino_101
+endif
+
+ifeq ($(BOARD),qemu_x86)
+BOARD_NAME ?= qemu_x86
+endif
+
+TYPE  ?= release
+JERRYHEAP ?= 16
+
+# Include functionality like regular expressions
+# check Jerry script documentation
+#
+# -cp
+# -cp_minimal
+# -cp_minimal-mem_stats
+# -mem_stats
+# -mem_stress_test
+
+ifndef ZEPHYR_BASE
+$(error Missing Zephyr base, did you source zephyr-env.sh? )
+endif
+
+VARIETY ?= -cp_minimal
+
+INTERM     = build/$(BOARD)/obj-$(BOARD)
+OUTPUT     = build/$(BOARD)
+
+CC = $(CROSS_COMPILE)gcc
+
+EXT_CFLAGS := -fno-asynchronous-unwind-tables -fno-omit-frame-pointer 
+EXT_CFLAGS += -fno-stack-protector -fno-strict-overflow -ffreestanding
+EXT_CFLAGS += -fno-reorder-functions -fno-defer-pop -fdata-sections
+EXT_CFLAGS += -ffunction-sections -fno-inline-functions
+
+# TODO @sergioamr Read the arch and cflags from zephyr
+
+ifeq ($(BOARD),qemu_x86)
+CONFIG_TOOLCHAIN_VARIANT = x86
+EXT_CFLAGS += -march=pentium
+else
+CONFIG_TOOLCHAIN_VARIANT = iamcu
+EXT_CFLAGS += -march=lakemont -mtune=lakemont -miamcu -msoft-float
+endif
+
+EXT_CFLAGS += -mpreferred-stack-boundary=2 -mno-sse
+
+EXT_CFLAGS += -Wall -Wno-format-zero-length -Wno-pointer-sign
+EXT_CFLAGS += -Werror=format -Werror=implicit-int -Wno-unused-but-set-variable
+EXT_CFLAGS += -Wno-main -Wno-strict-aliasing
+EXT_CFLAGS += -Wno-error=format=
+
+EXT_CFLAGS += $(TOOLCHAIN_CFLAGS)
+EXT_CFLAGS += $(LIB_INCLUDE_DIR)
+EXT_CFLAGS += -nostdinc -isystem $(shell $(CC) -print-file-name=include)
+EXT_CFLAGS += -isystem $(shell $(CC) -print-file-name=include-fixed)
+
+-include $(ZEPHYR_BASE)/boards/$(BOARD_NAME)/Makefile.board
+-include $(ZEPHYR_BASE)/scripts/Makefile.toolchain.$(ZEPHYR_GCC_VARIANT)
+
+ZEPHYR_LIBC_INC = $(subst -I,,$(TOOLCHAIN_CFLAGS))
+LIB_INCLUDE_DIR += -L $(CURDIR)/$(OUTPUT)
+
+EXTERNAL_LIB = $(INTERM)/lib$(TYPE).external$(VARIETY)-entry.a
+ZEPHYR_BIN = $(OUTPUT)/zephyr/zephyr.strip
+
+PREFIX = $(TYPE)$(VARIETY)
+LIBS = $(TYPE).external$(VARIETY)-entry $(PREFIX).jerry-core $(PREFIX).jerry-libm.lib $(TOOLCHAIN_LIBS)
+
+# TODO @sergioamr Change how we link the library to USER_LDFLAGS
+# https://gerrit.zephyrproject.org/r/#/c/2048/
+
+BUILD_CONFIG = O="$(OUTPUT)/zephyr" V=$(V) TOOLCHAIN_LIBS="$(LIBS)" LIB_INCLUDE_DIR="$(LIB_INCLUDE_DIR)"
+
+.PHONY: all
+all: jerry zephyr
+
+$(EXTERNAL_LIB):
+ifdef V
+	@echo "- JERRY SCRIPT -------------------------------------------------"
+endif
+	mkdir -p $(INTERM)
+	mkdir -p $(OUTPUT)
+	cmake -B$(INTERM) -H./ \
+	 -DENABLE_LTO=OFF \
```

Comment:
I just want to note that LTO provides a big improvement. At some point you should consider enabling it. All-in-one is also an alternative.